drivers/i2c/i2c_sim.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_sim_i2c
+
+#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(i2c_sim);
+
+#include <device.h>
+#include "i2c-priv.h"
+
+/* Structure declarations */
+
+struct i2c_sim_data {
+	bool slave_attached;
+	struct i2c_slave_config *slave_cfg;
+};
+
+struct i2c_sim_cfg {
+	uint32_t base;
+};
+
+static int i2c_sim_configure(struct device *dev, uint32_t dev_config)
+{
+	const struct i2c_sim_cfg *config = NULL;
+
+	/* Check for NULL pointers */
+	if (!dev) {
```

Comment:
This comment is not necessary, you could change it to `if (dev == NULL)`

Reply:
done